src/main/java/org/junit/runner/manipulation/Ordering.java

```diff
@@ -0,0 +1,158 @@
+package org.junit.runner.manipulation;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+
+import org.junit.runner.Description;
+import org.junit.runner.OrderWith;
+
+/**
+ * Reorders tests. An {@code Ordering} can reverse the order of tests, sort the
+ * order or even shuffle the order.
+ * 
+ * <p>In general you will not need to use a <code>Ordering</code> directly.
+ * Instead, use {@link org.junit.runner.Request#orderWith(Ordering)}.
+ *
+ * @since 4.13
+ */
+public abstract class Ordering {
+    private static final String CONSTRUCTOR_ERROR_FORMAT
+            = "Ordering class %s should have a public constructor with signature %s(Ordering.Context context)";
```

Comment:
One option would be to define a `Factory` interface inside of `Ordering`:

    public interface Factory {
        Ordering create(Context context);
    }

Then `OrderWith` would take in a `Class<? extends Ordering.Factory>` and the only requirement on `Factory` would be that it would need to have a public no-arg constructor.